### PR TITLE
Chore: use jest without grunt

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

Using grunt in CI makes the output not readable for frontend tests. This pr leaves out grunt and uses jest instead. One thing that is not going to run is [sasslint](https://github.com/sasstools/sass-lint) which I think is not a big pain as we don't use sass files anymore for new components.

Maybe it would make sense to remove the commands and the grunt dependencies but I would do it altogether when the last grunt command is migrated as well.

This is also skipping now `no-focus-convey-tests` and `no-only-tests` @aocenas is this something we want to check in the future?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "e2e:dev": "./e2e/start-and-run-suite dev",
     "jest": "jest --notify --watch",
     "lint": "eslint . --ext .js,.tsx,.ts --cache",
-    "jest-ci": "mkdir -p reports/junit && export JEST_JUNIT_OUTPUT_DIR=reports/junit && jest --ci --reporters=default --reporters=jest-junit -w ${TEST_MAX_WORKERS:-100%}",
+    "test": "mkdir -p reports/junit && export JEST_JUNIT_OUTPUT_DIR=reports/junit && jest --ci --reporters=default --reporters=jest-junit -w ${TEST_MAX_WORKERS:-100%}",
```
Comment (aknuds1) [inline package.json:17]: Are you sure you want the "test" script to use CI parameters? For example this script passes `--ci` to jest and makes a report directory. I'm not sure that the script should be the same for local testing and CI.

**Review by aknuds1 — COMMENTED** [review]
In addition to my code comment, I am wondering if the "jest" target should be removed from scripts/grunt/options/exec.js?

File: package.json (hunk 1 above)
Comment (zoltanbedi) [reply package.json:17]: Maybe I will do a poll about this but personally I've never used the test command locally. I used the jest command instead. Maybe change this back to jest-ci and rename the jest command to test?

Comment (aknuds1) [reply package.json:17]: I think it's best to follow the convention of having the -ci suffix for CI scripts. If you don't have a non-ci script for local usage, that's fine. I think it doesn't matter if it's called jest-ci or test-ci FWIW.

**Review by aknuds1 — APPROVED** [review]
LGTM

File: package.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
     "e2e": "./e2e/start-and-run-suite",
     "e2e:debug": "./e2e/start-and-run-suite debug",
     "e2e:dev": "./e2e/start-and-run-suite dev",
-    "jest": "jest --notify --watch",
+    "test": "jest --notify --watch",
```
Comment (aocenas) [inline package.json:15]: I would probably do something like `test` and `test:watch`. Personally I usually do `yarn test` just to make sure there are no errors in other parts of code but locally I just run the tests in the IDE and as. With `jest --watch` no tests are run until something changes which usually isn't what I personally want.

**Comment by aocenas** [comment]
> This is also skipping now no-focus-convey-tests and no-only-tests @aocenas is this something we want to check in the future?

So I definitely think these are important. Basically if you do `it.only()` in jest tests it skips all other tests and nothing is run anymore while giving you green both locally and in CI which can be hard to notice (we had that issue already where for few days no tests were run in CI and nobody noticed)

**Comment by zoltanbedi** [comment]
> > This is also skipping now no-focus-convey-tests and no-only-tests @aocenas is this something we want to check in the future?
> 
> So I definitely think these are important. Basically if you do `it.only()` in jest tests it skips all other tests and nothing is run anymore while giving you green both locally and in CI which can be hard to notice (we had that issue already where for few days no tests were run in CI and nobody noticed)

We can bring back the no only tests fairly easily with [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest/) (I know I love these plugins 😄) but what about the other one?

**Comment by aocenas** [comment]
@zoltanbedi the other one is the same thing just for Go. You have the same issue there as you can do something similar with Convey tests (just different syntax)

**Comment by aocenas** [comment]
But I think those can just be a separate steps in the pipeline and still run through grunt or something like that without much changes. Just put the jest tests outside of Grunt runner

File: contribute/developer-guide.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ The test suite consists of three types of tests: _Frontend tests_, _backend test
 We use [jest](https://jestjs.io/) for our frontend tests. Run them using Yarn:
 
 ```
-yarn jest
+yarn test
```
Comment (oddlittlebird) [inline contribute/developer-guide.md:90]: I think it was funnier before the change.

File: package.json (hunk 2 above)
Comment (zoltanbedi) [reply package.json:15]: you can still run all tests by pressing `a`

**Review by aocenas — APPROVED** [review]

File: .drone.yml [hunk 4]
```diff
@@ -66,6 +66,7 @@ steps:
 - name: test-backend
   image: grafana/build-container:1.2.28
   commands:
+  - yarn grunt no-focus-convey-tests
```
Comment (aknuds1) [inline .drone.yml:69]: What does `yarn grunt no-focus-convey-tests` do? Why should it run for backend tests?

Comment (aknuds1) [reply .drone.yml:69]: I think we can replace it with this:
```suggestion
  - [[ $(grep FocusConvey -R pkg | wc -l) -eq "0" ]] || exit 1
```

File: scripts/lib.star [hunk 5]
```diff
@@ -441,7 +441,9 @@ def test_backend_step():
             'lint-backend',
         ],
         'commands': [
-            # First execute non-integration tests in parallel, since it should be safe
+            # First make sure that there are no tests with FocusConvey
+            'yarn grunt no-focus-convey-tests',
```
Comment (aknuds1) [inline scripts/lib.star:445]: I think it's better if we avoid Yarn and Grunt :D
```suggestion
            '[[ $(grep FocusConvey -R pkg | wc -l) -eq "0" ]] || exit 1',
```

**Review by aknuds1 — APPROVED** [review]

File: scripts/lib.star [hunk 6]
```diff
@@ -441,7 +441,9 @@ def test_backend_step():
             'lint-backend',
         ],
         'commands': [
-            # First execute non-integration tests in parallel, since it should be safe
+            # First make sure that there are no tests with FocusConvey
+            '[[ $(grep FocusConvey -R pkg | wc -l) -eq "0" ]] || exit 1',
```
Comment (aknuds1) [inline scripts/lib.star:445]: This shell syntax should work under Bourne shell I think:
```suggestion
            '[ $(grep FocusConvey -R pkg | wc -l) -eq "0" ] || exit 1',
```

**Pull request merged**